### PR TITLE
4.x controller

### PR DESCRIPTION
**Pull request by ADmad**
Removed all deprecated code from `Controller` and components.

### CODE REVIEW

File: src/Controller/Controller.php [hunk 1]
```diff
@@ -155,31 +151,6 @@ class Controller implements EventListenerInterface, EventDispatcherInterface
...
-     * ```
-     * public $components = ['RequestHandler', 'Acl'];
-     * ```
-     *
-     * @var array
-     * @link https://book.cakephp.org/3.0/en/controllers/components.html
-     *
-     * @deprecated 3.0.0 You should configure components in your Controller::initialize() method.
-     */
-    public $components = [];
```
Comment (markstory) [inline src/Controller/Controller.php]: For this and `$helpers` I wonder if we should trigger an error if we see them in use? We could add a `__set` that emits a warning telling people what they should be doing instead? Right now post upgrade their app would not work and figuring out why may take some time.

File: src/Controller/Controller.php [hunk 2]
```diff
@@ -250,8 +213,6 @@ public function __construct(ServerRequest $request = null, Response $response =
 
         $this->initialize();
 
-        $this->_mergeControllerVars();
-        $this->_loadComponents();
```
Comment (markstory) [inline src/Controller/Controller.php]: We could put the new warning right here :smile:

**Comment by markstory** [comment]
Refs #11934

**Comment by ADmad** [comment]
@markstory Warning added as suggested.

File: tests/TestCase/View/ViewVarsTraitTest.php [hunk 3]
```diff
@@ -186,12 +186,10 @@ public function testUptoDateViewVars()
      */
     public function testViewOptionsGetsToBuilder()
     {
-        $this->deprecated(function () {
-            $this->subject->passedArgs = 'test';
-            $this->subject->createView();
-            $result = $this->subject->viewBuilder()->getOptions();
-            $this->assertEmpty($result);
-        });
+        $this->subject->passedArgs = 'test';
```
Comment (ADmad) [inline tests/TestCase/View/ViewVarsTraitTest.php]: Don't see the point of setting an unused property. Either it should be replaced by equivalent new method or the whole test method should be removed.

Comment (saeideng) [reply tests/TestCase/View/ViewVarsTraitTest.php]: >test that removed deprecated passedArgs is no longer passed to the view

Comment (saeideng) [reply tests/TestCase/View/ViewVarsTraitTest.php]: >should be removed.

:+1:

**Comment by markstory** [comment]
Nice work @ADmad :clap:

**Pull request merged**